tests/api-watch/test_diff_modtype_decl.ml

```diff
@@ -0,0 +1,98 @@
+open Api_watch
+open Test_helpers
+
+let%expect_test "Module types with both value and submodule changes" =
```

Comment:
This title is not very clear, what do you mean by that? What type of change are you testing?